### DISCUSSION TITLE
Is there a way to add comments in scripts?

### DISCUSSION THREAD

**Discussion by jbertovic** [nushell/nushell#4160] [Q&A] [answered]
Just wondering if there is a way to add comments in script files?

**Comment by jbertovic** [comment, marked as answer]
Okay...probably was a stupid question.  Playing around with different characters....noticed an example scripted used '#' to indicate a comment line and it works.  I couldn't find it in the documentation so ended up asking the question here.

best place for example scripts: [nu_scripts](https://github.com/nushell/nu_scripts)

**Reply by ghost** [reply]
> Okay...probably was a stupid question. 

I find this a great question. Thank you for posting it and the answer! 

> I couldn't find it in the documentation 

I have also had this struggle so often look for examples in `nu_scripts`. 
I think part of the broader problem is the core team and community are designing the semantics of the `nu` language as an open source project.
So the semantics are growing.

**Reply by kubouch** [reply]
Yes, comments are `#`. It's one of those things that seem so obvious you forget to document it :-).

Please, if you find something missing / inaccurate in the docs, file an issue or you can just hit the edit button on GitHub. We're not putting so much effort into the documentation since a lot of things are changing all the time. But we're getting close to merging the huge engine update and that would be a good time to update the documentation thoroughly.

**Reply by jbertovic** [reply]
Is it worth changing the docs until the new engine-q gets implemented?  We could add/modify them based on how the in shell help commands get worded.

I think it's great all of the changes to nushell and possibilities with dataframe/polars.  The difficult part as you've mentioned is trying to keep up with the changes.  I'm always up for contributing where I can.

**Reply by kubouch** [reply]
For example, the `#` is worth adding since that is not going to change with the update. But yeah, it's better to wait for the update before making significant changes to the official book.

I started documenting the engine-q specific changes here: https://github.com/nushell/engine-q/tree/main/docs . These pages are intended to be merged into the main book later.
